### PR TITLE
Mihajlokolesar#1599 automation testing test1,2,3

### PR DESCRIPTION
**Pull request by mihajlokolesar**
### Description:
to run java and mvn are required

to run without browserstack locally on some machine specific browser and corresponding driver is required
following command can be executed:
`mvn clean verify -Dbrowser=ch -Dwebdriver.chrome.driver= <path to selenium chrome web driver on the system> -Dbaseurl=<full base url off application, by default it's https://beta.bigneon.com/>`

`mvn clean verify -Dbrowser=ff -Dwebdriver.gecko.driver= <path to selenium firefox web driver on the system> -Dbaseurl=<full base url off application, by default it's https://beta.bigneon.com/>`

`mvn clean verify -Dbrowser=sf (on mac safari webdriver must be enabled and safari must be enabled to be controlled for automation) -Dbaseurl=<full base url off application, by default it's https://beta.bigneon.com/>`

to run test on browser stack
`mvn clean verify -P browserstack -Dusername=<browser_stack_username> -Dkey=<browser_stack_access_key> -Dbrowser=rm`
### Environment Variables
 * No change

### CODE REVIEW

File: selenium/qa/src/test/java/test/LoginStepsIT.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+	@Test(dataProvider = "user_credentials")
+	public void regularLogin(String username, String password) {
+		loginPage = new LoginPage(driver);
+		Assert.assertTrue(loginPage.login(username, password));
+
+	}
+	
+	@DataProvider(name = "user_credentials")
+	public static Object[][] data() {
+		return new Object[][] { { "testuser@mailnator.com", "test1111" } };
```
Comment (Krakaw) [inline selenium/qa/src/test/java/test/LoginStepsIT.java]:
```suggestion
		return new Object[][] { { "testuser@mailinator.com", "test1111" } };
```

File: selenium/qa/src/test/java/test/LoginWithFacebookStepsIT.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+		Assert.assertTrue(loginPage.isAtPage());
+		boolean isLogedIn = loginPage.loginWithFacebookUsingMail(username, password);
+		Assert.assertTrue(isLogedIn);
+		HomePage homePage = new HomePage(driver);
+		Assert.assertTrue(homePage.isAtPage());
+	}
+
+	@DataProvider(name = "user_fb_credentials")
+	public static Object[][] data() {
+		return new Object[][] { { "tusertrqa@gmail.com", "test/1111/" } };
```
Comment (Krakaw) [inline selenium/qa/src/test/java/test/LoginWithFacebookStepsIT.java]: Use environment variables?

File: selenium/qa/src/test/java/test/SignUpStepsIT.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+		}
+		Assert.assertTrue(retVal);
+	}
+
+	private Map<String, String> generateRandomUser() {
+		Random random = new Random();
+		Integer ran = random.nextInt(1000000);
+		String firstName = "seleniumtest";
+		String lastName = "qaselenium";
+		String emailAddress = firstName + ran + "@mailnator.com";
```
Comment (Krakaw) [inline selenium/qa/src/test/java/test/SignUpStepsIT.java]:
```suggestion
		String emailAddress = firstName + ran + "@mailinator.com";
```

File: selenium/qa/pom.xml [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+		<project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+		<java.version>1.8</java.version>
+		<selenium.version>3.9.1</selenium.version>
+		<testng.version>6.14.3</testng.version>
+		<java.mail.version>1.4.7</java.mail.version>
+		<maven-compiler-plugin.version>3.7.0</maven-compiler-plugin.version>
+		<maven-failsafe-plugin.version>2.21.0</maven-failsafe-plugin.version>
+		<browser>rm</browser>
+		<baseurl>https://beta.bigneon.com/</baseurl>
```
Comment (Krakaw) [inline selenium/qa/pom.xml]:
```suggestion
		<baseurl>https://develop.bigneon.com/</baseurl>
```

File: selenium/qa/src/test/java/test/LoginWithFacebookStepsIT.java (hunk 2 above)
Comment (mihajlokolesar) [reply selenium/qa/src/test/java/test/LoginWithFacebookStepsIT.java]: related to all data in data providers, can I suggest to move that in some properties file?

File: selenium/qa/src/test/java/config/DriverManager.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
+package config;
+
+import java.net.MalformedURLException;
+
+import org.openqa.selenium.WebDriver;
+
+public abstract class DriverManager {
+	
+	public static final String USERNAME = "mihajlokolesar1";
```
[the remark this answers is not in the document]
Comment (mihajlokolesar) [reply selenium/qa/src/test/java/config/DriverManager.java]: this is injected variable, I will remove this part of code. sorry

File: selenium/qa/src/test/java/test/LoginWithFacebookStepsIT.java (hunk 2 above)
Comment (Krakaw) [reply selenium/qa/src/test/java/test/LoginWithFacebookStepsIT.java]: That's perfect.

**Pull request merged**